### PR TITLE
Add configuration to restrict allowed user visibility modes

### PR DESCRIPTION
**Pull request by 6543**
Now that #16069 is merged, some sites may wish to enforce that users are all public, limited or private, and/or disallow users from becoming private.

This PR adds functionality and settings to constrain a user's ability to change their visibility.

### CODE REVIEW

File: modules/setting/service.go [hunk 1]
```diff
@@ -17,6 +17,8 @@ import (
 var Service struct {
 	DefaultUserVisibility                   string
 	DefaultUserVisibilityMode               structs.VisibleType
+	AllowedUserVisibilityModes              []string
+	AllowedUserVisibilityModesMap           map[structs.VisibleType]bool `ini:"-"`
```
Comment (zeripath) [inline modules/setting/service.go]: As VisibleType is an integer with complete coverage just use an bool array.

File: modules/setting/service.go [hunk 2]
```diff
@@ -75,6 +75,16 @@ var Service struct {
 	} `ini:"service.explore"`
 }
```
Comment (zeripath) [inline modules/setting/service.go:78]: Why not change this to:

```suggestion
} {
  AllowedUserVisibilityModesSlice: []bool{true, true, true},
}
```

Comment (zeripath) [reply modules/setting/service.go:78]: Then you could drop all of those changes in the tests

**Comment by zeripath** [comment]
sorry for the broken suggestion

**Comment by 6543** [comment]
no worry this pull should be now ready :)

**Review by zeripath — APPROVED** [review]

**Review by noerw — APPROVED** [review]

**Pull request merged**